### ISSUE TITLE
proposal: add a parseutils sub-package under x/net/html

### ISSUE THREAD

**Issue by srinathh** [golang/go#16419]
When using `x/net/html` to parse a HTML Spec to generate Go code in the [colornames](http://godoc.org/golang.org/x/image/colornames) package, we wrote several [useful and fairly generalised functions](https://github.com/golang/image/blob/master/colornames/gen.go) to interact with HTML parse trees such as a generalised node matching function type, match function generators for nodes by atom type & attributes and functions to recursively search and find matching nodes in the parse tree.

I have since then found these functions very useful in several other projects when using the experimental HTML parser. Therefore I was wondering if there is merit in providing these types of functions as a `parseutils` sub-package in the experimental repo. Happy to work on it.

**Comment by adg** [comment]
It seems this was declined in error. But this issue is blocked on the greater issue #17244

**Comment by rsc** [comment]
On hold for #17244.
